packages/fx-core/src/plugins/resource/bot/v2/plugin.ts

```diff
@@ -303,23 +283,65 @@ export class TeamsBotV2Impl {
     throw new Error("Invalid programming language");
   }
 
-  private resolveScenarios(
-    actRoles: PluginActRoles[],
-    triggers: BotTrigger[],
-    hostType: HostType
-  ): string[] {
-    const scenarios: string[] = [];
-    if (hostType === HostType.Function) {
-      if (actRoles.includes(PluginActRoles.Notification)) {
-        scenarios.push(TemplateProjectsScenarios.NOTIFICATION_FUNCTION_BASE_SCENARIO_NAME);
-        triggers.map((trigger) => scenarios.push(TriggerTemplateScenarioMappings[trigger]));
+  private resolveScenariosForMessageExtension(
+    ctx: Context,
+    inputs: Inputs,
+    botGroupScenarios: Set<string>,
+    msgextGroupScenarios: Set<string>
+  ): void {
+    const isM365 = ctx.projectSetting?.isM365;
+    if (isM365) {
+      botGroupScenarios.add(TemplateProjectsScenarios.M365_SCENARIO_NAME);
+    } else {
+      if (inputs.platform === Platform.VS) {
+        msgextGroupScenarios.add(TemplateProjectsScenarios.DEFAULT_SCENARIO_NAME);
+      } else {
+        botGroupScenarios.add(TemplateProjectsScenarios.DEFAULT_SCENARIO_NAME);
       }
     }
-    if (hostType === HostType.AppService) {
-      // TODO: support command & respond bot and notification bot
-      scenarios.push(TemplateProjectsScenarios.DEFAULT_SCENARIO_NAME);
+  }
+
+  private resolveScenariosForBot(
+    ctx: Context,
+    inputs: Inputs,
+    botGroupScenarios: Set<string>,
+    msgextGroupScenarios: Set<string>
+  ): void {
+    const botScenarios = inputs?.[AzureSolutionQuestionNames.Scenarios];
+    if (!botScenarios) {
+      botGroupScenarios.add(TemplateProjectsScenarios.DEFAULT_SCENARIO_NAME);
+    } else {
+      botScenarios.map((scenario: string) => {
+        switch (scenario) {
+          case BotScenario.CommandAndResponseBot:
+            botGroupScenarios.add(TemplateProjectsScenarios.COMMAND_AND_RESPONSE_SCENARIO_NAME);
+            break;
+          case BotScenario.NotificationBot:
+            const hostType = this.resolveHostType(inputs);
+            if (hostType === HostTypes.AZURE_FUNCTIONS) {
+              botGroupScenarios.add(
+                TemplateProjectsScenarios.NOTIFICATION_FUNCTION_BASE_SCENARIO_NAME
+              );
+              const triggers = this.resolveTriggers(inputs);
+              triggers.map((trigger) =>
+                botGroupScenarios.add(TriggerTemplateScenarioMappings[trigger])
```

Comment:
Since this is also another mapping definition, we can keep all mapping into one separate file to make us easy to config it for business.

Reply:
Moved these maps to mapping.ts.